stdlib/Unicode/src/Unicode.jl

```diff
@@ -89,4 +89,45 @@ letter combined with an accent mark is a single grapheme.)
 """
 graphemes(s::AbstractString) = Base.Unicode.GraphemeIterator{typeof(s)}(s)
 
+const EMOJI_RANGES = [
+    0x1F600:0x1F64F,  # Emoticons
+    0x1F300:0x1F5FF,  # Misc Symbols and Pictographs
+    0x1F680:0x1F6FF,  # Transport and Map
+    0x02600:0x026FF,  # Misc symbols
+    0x02700:0x027BF,  # Dingbats
+    0x0FE00:0x0FE0F,  # Variation Selectors
+    0x1F900:0x1F9FF,   # Supplemental Symbols and Pictographs
```

Comment:
Should this be generated from https://unicode.org/Public/emoji/13.1/emoji-sequences.txt somehow?     Or at least have a test that checks that everything from https://unicode.org/Public/emoji/13.1/emoji-sequences.txt is in this set?

Reply:
That's a good idea. I'll get to work on this. Is it ok to include a local copy of this file in the /Unicode/test/ directory?

Reply:
Can't we just download it during the test?

Reply:
We can but it may make CI flaky. Let's try that and see how it goes. If it's flaky we may need to cache a copy but that can be dealt with separately.

Reply:
I implemented y'all's suggestions, so now we download https://www.unicode.org/Public/13.0.0/ucd/emoji/emoji-data.txt from the unicode website and use that to generate a const containing emoji ranges, which are then tested against. In the tests, we download the https://unicode.org/Public/emoji/13.1/emoji-sequences.txt and https://www.unicode.org/Public/emoji/13.1/emoji-zwj-sequences.txt files, which we use to build a list of all valid emoji, which we then test the isemoji function against.

Reply:
I agree that we shouldn't return `true` for currently unassigned codepoints, but we can check for that simply by returning `false` if `Unicode.category_code(char) == Unicode.UTF8PROC_CATEGORY_CN`